src/main/java/org/apache/sling/distribution/journal/kafka/KafkaEndpoint.java

```diff
@@ -36,4 +37,17 @@
     @AttributeDefinition(name = "Kafka Default API Timeout",
             description = "Kafka Default API Timeout in ms.")
     int kafkaDefaultApiTimeout() default 60000;
+    
+    @AttributeDefinition(name = "Security protocol",
+            description = "e.g. SASL_SSL")
+    SecurityProtocol securityProtocol() default SecurityProtocol.PLAINTEXT;
+    
+    @AttributeDefinition(name = "Sasl mechanism",
+            description = "e.g. PLAIN")
+    String saslMechanism() default "";
+    
+    @AttributeDefinition(name = "Sasl jaas config",
+            description = "")
+    String saslJaasConfig() default "";
```

Comment:
The [default sasl.jaas.config](https://kafka.apache.org/documentation/#api) seems to be `null`, we should keep the same default here as well.

Reply:
Ok .. but I need to make it optional.

Reply:
I have not found a way to make this optional. So I leave "" as default in the annotation and set it to null in this case.

Reply:
It should be possible to have a `null` default by simply avoiding specifying any `default`. It would cause issues with the default converter used in the tests, I have opened FELIX-6137.